src/streaming/protection/servers/PlayReady.js

```diff
@@ -43,6 +45,74 @@ function PlayReady() {
 
     let instance;
 
+    const soap = 'http://schemas.xmlsoap.org/soap/envelope/';
+
+    function uintToString(arrayBuffer) {
+        let encodedString = String.fromCharCode.apply(null, new Uint8Array(arrayBuffer));
+        let decodedString = decodeURIComponent(escape(encodedString));
+        return decodedString;
+    }
+
+    function parseServerResponse(serverResponse) {
+        if (window.DOMParser) {
+            let stringResponse = uintToString(serverResponse);
+            let parser = new window.DOMParser();
+            let xmlDoc = parser.parseFromString(stringResponse, 'text/xml');
+            let enveloppe = xmlDoc ? xmlDoc.getElementsByTagNameNS(soap, 'Envelope')[0] : null;
```

Comment:
Aren't we too optimistic assuming XML is going to be well-formed and with the right structure? Makes sense checking if the elements we are looking for really exist?

Note: minor thing, there is a typo in name variable (it should be envelope).

Reply:
I thinks it is OK, because if there is no soap:Envelope or anything else, we return the server response not modified. 
There is no exception when we parseFromString something that is not xml, we get a xml document with a parserror tag.
The only thing we have to return is null when there is a <soap:Fault> tag in the server response.
There is a unit test that mocks a playready invalid license (test/unit/streaming.protection.servers.PlayReady.js with /unit/data/licence/playredayInvalidLicence.txt mock response) and it passes... 
Have I missed something wrong ?

I have fixed the typo in the variable name 'envelope'